daemon/daemon.go

```diff
@@ -795,6 +808,12 @@ func (daemon *Daemon) Shutdown() error {
 	// Keep mounts and networking running on daemon shutdown if
 	// we are to keep containers running and restore them.
 
+	// If we are part of a cluster, clean up cluster's stuff
+	if daemon.clusterProvider != nil {
+		logrus.Debugf("start clean shutdown of cluster resources...")
+		daemon.DaemonLeavesCluster()
```

Comment:
Yes, as long as it is being called before `netController.Stop()`.
Currently, on daemon graceful shutdown, we were skipping all the cluster resources cleanup.
I feel having this gracefully cleanup what done during leave is the proper thing to do..
Although I am not sure yet if better to invoke this before or after the containers shutdown.
@tonistiigi input here would help.